src/iot/explorer/iotCertificateNode.ts

```diff
@@ -0,0 +1,191 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as moment from 'moment'
+import { ChildNodePage } from '../../awsexplorer/childNodeLoader'
+import { IotThing, IotClient, IotCertificate } from '../../shared/clients/iotClient'
+import { ext } from '../../shared/extensionGlobals'
+import { AWSResourceNode } from '../../shared/treeview/nodes/awsResourceNode'
+import { AWSTreeNodeBase } from '../../shared/treeview/nodes/awsTreeNodeBase'
+import { ErrorNode } from '../../shared/treeview/nodes/errorNode'
+import { LoadMoreNode } from '../../shared/treeview/nodes/loadMoreNode'
+import { PlaceholderNode } from '../../shared/treeview/nodes/placeholderNode'
+import { makeChildrenNodes } from '../../shared/treeview/treeNodeUtilities'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { ChildNodeLoader } from '../../awsexplorer/childNodeLoader'
+import { Workspace } from '../../shared/vscode/workspace'
+import { inspect } from 'util'
+import { getLogger } from '../../shared/logger'
+import { IotNode } from './iotNodes'
+import { IotThingFolderNode } from './iotThingFolderNode'
+import { IotCertsFolderNode } from './iotCertFolderNode'
+import { IotThingNode } from './iotThingNode'
+import { IotPolicyNode } from './iotPolicyNode'
+
+const CONTEXT_BASE = 'awsIotCertificateNode'
+
+const STATUS_REVOKED = 'REVOKED'
+const STATUS_ACTIVE = 'ACTIVE'
+const STATUS_INACTIVE = 'INACTIVE'
+
+/**
+ * Shamelessly stolen from S3.
+ */
+const S3_DATE_FORMAT = 'll LTS [GMT]ZZ'
+
+/**
+ * Represents an IoT Certificate that may have either a Thing Node or the
+ * Certificate Folder Node as a parent.
+ */
+export class IotCertificateNode extends AWSTreeNodeBase implements AWSResourceNode {
+    public constructor(
+        public readonly certificate: IotCertificate,
+        public readonly parent: IotCertsFolderNode | IotThingNode,
+        public readonly iot: IotClient,
+        collapsibleState: vscode.TreeItemCollapsibleState,
+        protected readonly workspace = Workspace.vscode()
+    ) {
+        //Show only 8 characters in the explorer instead of the full 64. The entire
+        //ID can be copied from the context menu or viewed when hovered over.
```

Comment:
Shortening this makes sense. Is this a standard convention though? Does the console do anything like this?

Reply:
I don't think this is standard. The console just shows however many characters will fit into a single line given the width of the browser